File: start-echo.py
#!/usr/bin/env python3
"""
Python script to launch EchoBooster application
This script opens a terminal and runs the EchoBooster application
"""

import os
import sys
import subprocess
import platform
from pathlib import Path


def find_echo_booster_executable():
    """Find the EchoBooster executable in common locations"""
    possible_paths = [
        # Current directory
        "./EchoBooster.exe",
        "./EchoBooster",
        
        # Subdirectories
        "./EchoBooster/bin/Release/net6.0-windows/EchoBooster.exe",
        "./EchoBooster/bin/Debug/net6.0-windows/EchoBooster.exe",
        "./EchoBooster/bin/Release/net6.0/EchoBooster",
        "./EchoBooster/bin/Debug/net6.0/EchoBooster",
        
        # Parent directory
        "../EchoBooster.exe",
        "../EchoBooster",
        
        # Common build locations
        "./bin/Release/net6.0-windows/EchoBooster.exe",
        "./bin/Debug/net6.0-windows/EchoBooster.exe",
        "./bin/Release/net6.0/EchoBooster",
        "./bin/Debug/net6.0/EchoBooster",
    ]
    
    script_dir = Path(__file__).parent.absolute()
    
    for path in possible_paths:
        full_path = script_dir / path
        if full_path.exists():
            return str(full_path)
    
    return None


def open_terminal_and_run():
    """Open a terminal and run the EchoBooster application"""
    print("Starting EchoBooster Application...")
    
    # Try to find the executable
    exe_path = find_echo_booster_executable()
    
    if exe_path:
        print(f"Found EchoBooster at: {exe_path}")
        print("Launching EchoBooster...")
        
        try:
            if platform.system() == "Windows":
                # On Windows
                subprocess.Popen([exe_path], shell=True)
            elif platform.system() == "Darwin":  # macOS
                # On macOS, try to open in Terminal
                subprocess.Popen(['open', '-a', 'Terminal', exe_path])
            else:  # Linux
                # On Linux, try different terminal emulators
                terminals = ['gnome-terminal', 'konsole', 'xfce4-terminal', 'xterm']
                terminal_found = False
                
                for terminal in terminals:
                    if subprocess.run(['which', terminal], capture_output=True).returncode == 0:
                        subprocess.Popen([terminal, '-e', exe_path])
                        terminal_found = True
                        break
                
                if not terminal_found:
                    # Fallback: run directly
                    subprocess.Popen([exe_path])
            
            print("EchoBooster launched successfully!")
        except Exception as e:
            print(f"Error launching EchoBooster: {e}")
            return False
    else:
        print("EchoBooster executable not found!")
        print("Attempting to build and run from source...")
        
        # Try to run from source using dotnet if available
        try:
            result = subprocess.run(['dotnet', '--version'], capture_output=True, text=True)
            if result.returncode == 0:
                # DotNet is available, try to run the project
                script_dir = Path(__file__).parent.absolute()
                project_path = script_dir / "EchoBooster" / "EchoBooster.csproj"
                
                if project_path.exists():
                    print("Found EchoBooster project, building and running...")
                    subprocess.Popen(['dotnet', 'run', '--project', 'EchoBooster'], cwd=script_dir)
                    return True
                else:
                    print("EchoBooster project not found.")
            else:
                print("DotNet SDK not found. Please install .NET SDK to run from source.")
        except FileNotFoundError:
            print("DotNet SDK not found. Please install .NET SDK to run from source.")
        
        # Try to open a terminal for manual operation
        try:
            if platform.system() == "Windows":
                subprocess.Popen(['cmd'], shell=True)
            elif platform.system() == "Darwin":
                subprocess.Popen(['open', '-a', 'Terminal', '.'])
            else:  # Linux
                terminals = ['gnome-terminal', 'konsole', 'xfce4-terminal', 'xterm']
                terminal_found = False
                
                for terminal in terminals:
                    if subprocess.run(['which', terminal], capture_output=True).returncode == 0:
                        subprocess.Popen([terminal])
                        terminal_found = True
                        break
                
                if not terminal_found:
                    print("No known terminal found. Please open a terminal manually.")
                    return False
            
            print("Terminal opened. Please navigate to the EchoBooster directory and run 'dotnet run'.")
            return True
        except Exception as e:
            print(f"Error opening terminal: {e}")
            return False
    
    return True


def main():
    """Main function"""
    print("EchoBooster Launcher")
    print("=" * 20)
    
    success = open_terminal_and_run()
    
    if success:
        print("\nOperation completed successfully.")
    else:
        print("\nOperation failed.")
    
    # Wait for user input before exiting
    input("\nPress Enter to exit...")


if __name__ == "__main__":
    main()

File: start-echo.js
/**
 * Node.js script to launch EchoBooster application
 * This script opens a terminal and runs the EchoBooster application
 */

const { spawn, exec } = require('child_process');
const { platform } = require('os');
const { join, dirname } = require('path');
const fs = require('fs');

// Get the directory where this script is located
const scriptDir = __dirname;

/**
 * Find the EchoBooster executable in common locations
 */
function findEchoBoosterExecutable() {
    const possiblePaths = [
        // Current directory
        join(scriptDir, 'EchoBooster.exe'),
        join(scriptDir, 'EchoBooster'),
        
        // Subdirectories
        join(scriptDir, 'EchoBooster', 'bin', 'Release', 'net6.0-windows', 'EchoBooster.exe'),
        join(scriptDir, 'EchoBooster', 'bin', 'Debug', 'net6.0-windows', 'EchoBooster.exe'),
        join(scriptDir, 'EchoBooster', 'bin', 'Release', 'net6.0', 'EchoBooster'),
        join(scriptDir, 'EchoBooster', 'bin', 'Debug', 'net6.0', 'EchoBooster'),
        
        // Parent directory
        join(scriptDir, '..', 'EchoBooster.exe'),
        join(scriptDir, '..', 'EchoBooster'),
        
        // Common build locations
        join(scriptDir, 'bin', 'Release', 'net6.0-windows', 'EchoBooster.exe'),
        join(scriptDir, 'bin', 'Debug', 'net6.0-windows', 'EchoBooster.exe'),
        join(scriptDir, 'bin', 'Release', 'net6.0', 'EchoBooster'),
        join(scriptDir, 'bin', 'Debug', 'net6.0', 'EchoBooster'),
    ];

    for (const path of possiblePaths) {
        if (fs.existsSync(path)) {
            return path;
        }
    }

    return null;
}

/**
 * Open a terminal and run the EchoBooster application
 */
function openTerminalAndRun() {
    console.log('Starting EchoBooster Application...');

    // Try to find the executable
    const exePath = findEchoBoosterExecutable();

    if (exePath) {
        console.log(`Found EchoBooster at: ${exePath}`);
        console.log('Launching EchoBooster...');

        try {
            if (platform() === 'win32') {
                // On Windows
                spawn(exePath, [], { detached: true, stdio: 'ignore' });
            } else if (platform() === 'darwin') {  // macOS
                // On macOS, try to open in Terminal
                spawn('open', ['-a', 'Terminal', exePath]);
            } else {  // Linux
                // On Linux, try different terminal emulators
                const terminals = ['gnome-terminal', 'konsole', 'xfce4-terminal', 'xterm'];
                let terminalFound = false;

                for (const terminal of terminals) {
                    exec(`which ${terminal}`, (error, stdout, stderr) => {
                        if (!error && stdout.trim()) {
                            spawn(terminal, ['-e', exePath]);
                            terminalFound = true;
                            return;
                        }
                    });
                }

                if (!terminalFound) {
                    // Fallback: run directly
                    spawn(exePath);
                }
            }

            console.log('EchoBooster launched successfully!');
            return true;
        } catch (error) {
            console.log(`Error launching EchoBooster: ${error.message}`);
            return false;
        }
    } else {
        console.log('EchoBooster executable not found!');
        console.log('Attempting to build and run from source...');

        // Try to run from source using dotnet if available
        exec('dotnet --version', (error, stdout, stderr) => {
            if (!error) {
                // DotNet is available, try to run the project
                const projectPath = join(scriptDir, 'EchoBooster', 'EchoBooster.csproj');

                if (fs.existsSync(projectPath)) {
                    console.log('Found EchoBooster project, building and running...');
                    spawn('dotnet', ['run', '--project', 'EchoBooster'], {
                        cwd: scriptDir,
                        stdio: 'inherit'
                    });
                } else {
                    console.log('EchoBooster project not found.');
                }
            } else {
                console.log('DotNet SDK not found. Please install .NET SDK to run from source.');
            }
        });

        // Try to open a terminal for manual operation
        try {
            if (platform() === 'win32') {
                spawn('cmd', [], { detached: true, stdio: 'ignore' });
            } else if (platform() === 'darwin') {  // macOS
                spawn('open', ['-a', 'Terminal', scriptDir]);
            } else {  // Linux
                const terminals = ['gnome-terminal', 'konsole', 'xfce4-terminal', 'xterm'];
                let terminalFound = false;

                for (const terminal of terminals) {
                    exec(`which ${terminal}`, (error, stdout, stderr) => {
                        if (!error && stdout.trim()) {
                            spawn(terminal, [], { detached: true, stdio: 'ignore' });
                            terminalFound = true;
                            return;
                        }
                    });
                }

                if (!terminalFound) {
                    console.log('No known terminal found. Please open a terminal manually.');
                    return false;
                }
            }

            console.log('Terminal opened. Please navigate to the EchoBooster directory and run \'dotnet run\'.');
            return true;
        } catch (error) {
            console.log(`Error opening terminal: ${error.message}`);
            return false;
        }
    }

    return true;
}

/**
 * Main function
 */
function main() {
    console.log('EchoBooster Launcher');
    console.log('='.repeat(20));

    const success = openTerminalAndRun();

    if (success) {
        console.log('\nOperation completed successfully.');
    } else {
        console.log('\nOperation failed.');
    }

    // Wait for user input before exiting (in a real implementation, you might want to handle this differently)
    process.stdin.setEncoding('utf8');
    process.stdin.on('readable', () => {
        const chunk = process.stdin.read();
        if (chunk !== null) {
            process.exit(0);
        }
    });

    console.log('\nPress Enter to exit...');
}

// Run the main function
main();